app/jobs/delete_user_job.rb

```diff
@@ -0,0 +1,35 @@
+class DeleteUserJob < ApplicationJob
+  queue_as :default
+
+  def perform(user_attributes)
+    user = User.new(user_attributes)
+    family = Family.find(user.family_id)
+
+    if user.isMember
+      other_family_users = User.where(family_id: user.family_id).where.not(id: user.id).count
+
+      ActiveRecord::Base.transaction do
+        # this is true for our demo user but should not normally happen
+        if other_family_users == 0
+          if family != nil
+            family.destroy
+          end
+        end
+      end
+    elsif user.isAdmin
+      other_admins = User.where(family_id: user.family_id, role: "admin").where.not(id: user.id).count
+
+      if other_admins == 0
+        ActiveRecord::Base.transaction do
+          
+          if family != nil
+            family.destroy
+          end
+          # since other users are now invalid because of the orphaned family -> delete users as well
+          users = User.where(family_id: user.family_id)
+          users.destroy_all
```

Comment:
While I think my comment above (passing in the user instance) will simplify all of this logic a bit, we should move a lot of this method into the models themselves.  As an example:

```rb
other_admins = User.where(family_id: user.family_id, role: "admin").where.not(id: user.id).count
```

This should likely be a scope or method on `Family` so that we can call:

```rb
other_admins_count = Current.family.admins.where.not(id: user.id).count
```

Reply:
FYI - https://github.com/maybe-finance/maybe/issues/675#issuecomment-2081051730

Reply:
Gotcha.